inspirehep/records/marshmallow/literature/base.py

```diff
@@ -124,3 +129,163 @@ def get_ui_display(self, data):
+    author_count = fields.Method("populate_author_count")
+    authors = fields.Method("preprocess_authors")
+    bookautocomplete = fields.Method("populate_bookautocomplete")
+    earliest_date = fields.Method("populate_earliest_date")
+    facet_inspire_doc_type = fields.Method("populate_inspire_document_type")
+    facet_author_name = fields.Method("populate_facet_author_name")
+    id = fields.UUID(dump_only=True)
+    thesis_info = fields.Nested(ThesisInfoSchemaForESV1, dump_only=True)
+
+    def populate_abstract_source_suggest(self, record):
+        """Prepares record for ``abstract_source_suggest`` field."""
+        abstracts = record.get("abstracts", [])
+
+        for abstract in abstracts:
+            source = abstract.get("source")
+            if source:
+                abstract.update({"abstract_source_suggest": {"input": source}})
+        return abstracts
+
+    def populate_earliest_date(self, record):
+        """Prepares record for ``earliest_date`` field."""
+        earliest_date_found = record.get("earliest_date", "")
+        date_paths = [
+            "preprint_date",
+            "thesis_info.date",
+            "thesis_info.defense_date",
+            "publication_info.year",
+            "legacy_creation_date",
+            "imprints.date",
+        ]
+
+        dates = [
+            str(el)
+            for el in chain.from_iterable(
+                [force_list(record.get_value(path)) for path in date_paths]
+            )
+        ]
+        if dates:
+            result = earliest_date(dates)
+            if result:
+                earliest_date_found = result
+        return earliest_date_found
+
+    def populate_author_count(self, record):
+        """Prepares record for ``author_count`` field."""
+        authors = record.get("authors", [])
+
+        authors_excluding_supervisors = [
+            author
+            for author in authors
+            if "supervisor" not in author.get("inspire_roles", [])
+        ]
+        return len(authors_excluding_supervisors)
+
+    def preprocess_authors(self, record):
+        """Preprocess authors by adding ``full_name_normalized``
+        field and generating name variations"""
+        processed_authors = []
+        for index, author in enumerate(record.get("authors", [])):
+            author = self.prepare_author_full_name_unicode_normalized(author)
+            author = self.populate_name_variations_for_author(author)
+            processed_authors.append(author)
+        return processed_authors
+
+    @staticmethod
+    def prepare_author_full_name_unicode_normalized(author):
+        """Prepares data for ``author.full_name_normalized`` field."""
+        full_name = str(author["full_name"])
+        author.update(
+            {"full_name_unicode_normalized": normalize("NFKC", full_name).lower()}
+        )
+        return author
+
+    def populate_inspire_document_type(self, record):
```

Comment:
ditto

Reply:
I don't see how I could populate this field with nested. It requires access to `document_type`, `publication_type` and `refereed` fields from record itself.